reddit-background

```diff
@@ -139,10 +147,14 @@ class Desktop(object):
 
         random.shuffle(image_urls)
 
-        for image_url in image_urls:
+        log('Number of images to download: {0}'.format(NUMBER_OF_IMAGES))
+        number_downloaded = 0
+        while (number_downloaded < NUMBER_OF_IMAGES):
             try:
-                return _download_image(image_url)
+                _download_image(image_urls[number_downloaded])
+                number_downloaded += 1
             except URLOpenError:
+                number_downloaded += 1
```

Comment:
Ah yes, makes sense. It was a quick fix after it was caught in a endless loop trying to download an unavailable image